src/plugins/Polls/Services.tsx

```diff
@@ -0,0 +1,187 @@
+import Gun from 'gun'
+import 'gun/lib/then.js'
+import { Result, Err, Ok } from 'ts-results'
+import { gunServers } from '../../network/gun-servers'
+import type { PollMetaData } from './types'
+import { PluginMessageCenter } from '../PluginMessages'
+
+const gun = new Gun(gunServers)
+
+interface NewPollProps {
+    sender?: string | undefined
+    id?: string | undefined
+    question: string
+    options: Object
+    start_time: Date
+    end_time: Date
+}
+
+enum Reason {
+    NoPermission,
+    InvalidURL,
+    FetchFailed,
+}
+
+export async function createNewPoll(poll: NewPollProps): Promise<Result<NewPollProps, [Reason, Error?]>> {
+    const { id, sender, question, options, start_time, end_time } = poll
+
+    let results = {}
+    for (let i = 0; i < Object.values(options).length; i = i + 1) {
+        results = {
+            ...results,
+            [i]: 0,
+        }
+    }
+
+    const poll_item = {
```

Comment:
Please annotate it with something like `const poll_item: Type` if you already defined the `Type` in another place.